### DISCUSSION TITLE
Review of Number Compliance Manager (Office Hours: Thu Dec 3)

### DISCUSSION THREAD

**Discussion by myhlibovych** [twilio-labs/paste#966] [Office hours] [answered]
**Background:** Tool for reviewing non-compliant numbers to make them compliant in bulk
**Stage of design:** Hi-Fi Prototyping
**Link to prototype:** https://twilio.invisionapp.com/share/EHZGRAJQ49F

I'm using few components that I'm not sure how to align with Paste:
 — table with pagination
 — filters next to the table
 — confirmation in overlay
I'd like to review whole flow and get a green light for everything that's in the guidelines and list of what should I change to be compliant with Paste.

**Comment by serifluous** [comment, marked as answer]
Overall, this iteration feels much cleaner than the previous! Removing the containers seemed to help a lot.

> — table with pagination

We have Table and Pagination components that are in development. Engineers on the Numbers team have been involved in giving feedback, so if you make a note when you review the work with them that the components in your mockup should be swapped to the Paste components in development, this should be fine 👍

> — filters next to the table

The group of checkboxes should have a form label. Without it, the filter checkbox group seems a little _too_ related to the checkboxes in the table, almost like they're part of the same list. 

I left a few more comments in the Invision.

> — confirmation in overlay

Which confirmation do you mean? [This one in a modal](https://twilio.invisionapp.com/share/EHZGRAJQ49F#/screens/438055777) or [this toast](https://twilio.invisionapp.com/share/EHZGRAJQ49F#/screens/438055778)?

If the assignment is scheduled when a customer gets to this step of the modal, I'd suggest adding the toast to this screen rather than when they exit out of the modal OR make it much clearer on this step that the assignment was a success (with a green checkmark or something).

Toasts also have a [fixed width and will appear in the top right corner](https://paste.twilio.design/components/toast).

Some other notes:

- **Filters above the table:** I _think_ I remember from the research you shared that customers wanted to see a broad overview of all their non-compliant numbers. Would there be any value in having a "reset filters" option when you've selected a bunch of filters ([like this state](https://twilio.invisionapp.com/share/EHZGRAJQ49F#/screens/438055771)) and want to go back to seeing all non-compliant numbers? (Ignore me if there's already another way to do this or this isn't an issue)
- **Filters above the table:** Something else to keep in mind: If the results reload every time you make a change in one of the filters, make sure to test that the page is performant with your engineers. You could also try creating a test page with them in a [CodeSandbox](https://codesandbox.io/s/paste-starter-kit-rj7yy), just to get a feel for the interaction since it's hard to tell through static mockups. It could potentially get irritating if a page slows down a lot or page content is changing a lot before a customer finishes selecting through 5 form fields.
